libqtile/widget/base.py

```diff
@@ -279,6 +280,15 @@ def draw(self):
         """
         raise NotImplementedError
 
+    def draw_decorations(self):
```

Comment:
Should this also be in `Bar`? Some users have asked for single borders on the bar and this would be a way to achieve that.

Reply:
I think the answer to this is that it can't be done yet. The reason being is that all widgets (including spacers) are drawn over the bar so, without transparency, there's no point decorating the bar.

I do want to look at the bar transparency point separately...

Reply:
I don't it's important to do it in this PR so don't sweat it.

Reply:
Sorry, yes, I meant I'm hoping to look at it in a different branch but there's a bit more of a learning curve on that one for me.